src/main/java/io/vertx/httpproxy/impl/CacheImpl.java

```diff
@@ -1,23 +1,54 @@
 package io.vertx.httpproxy.impl;
 
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
 import io.vertx.httpproxy.cache.CacheOptions;
 import io.vertx.httpproxy.spi.cache.Cache;
+import io.vertx.httpproxy.spi.cache.Resource;
 
+import java.util.HashMap;
 import java.util.LinkedHashMap;
+import java.util.LinkedList;
 import java.util.Map;
 
 /**
  * Simplistic implementation.
  */
-public class CacheImpl<K, V> extends LinkedHashMap<K, V> implements Cache<K, V> {
+public class CacheImpl implements Cache {
 
   private final int maxSize;
+  private final LinkedHashMap<String, Resource> data;
 
   public CacheImpl(CacheOptions options) {
     this.maxSize = options.getMaxSize();
+    this.data = new LinkedHashMap<>() {
```

Comment:
If the cache is shared, it must be safe to use from different threads concurrently. Perhaps using `Collections.synchronizedMap()`